econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -938,7 +1014,7 @@ def whatif(self, X, Xnew, feature_index, y):
 
         return inf
 
-    def _whatif_dict(self, X, Xnew, feature_index, y, alpha=0.1):
+    def whatif(self, X, Xnew, feature_index, y, *, alpha=0.1):
```

Comment:
I think it would be good if the user passes a pandas df, that we maintain the same index in our results that we return, so that they can potentially do subsequent merging later on. 

Currently we get rid of the index and return a df with a 0, 1, 2, ... index